### PR TITLE
mesa: bump to 21.0.3

### PR DESCRIPTION
**Pull request by kenmays**
**Warning: Merge only for Haiku R1B3 and nightly builds.. Built and unit tested on Haiku R1B2 and latest nightly build.**

Mesa 21.0.3 implements the OpenGL 4.6 API and Vulkan 1.2 API.
- Stable release
- Compiled and packaged fine on Haiku hrev54972 x86 and x86_64 platforms.
- Unit tested. Only 7 failed out of 74 unit tests - minor Haiku GL type symbol export check errors.
- Works with Blender 2.92, Dragengine 1.4, and Godot 3.3

### CODE REVIEW

**Comment by kallisti5** [comment]
looks good to me.

There are several major issues which prevented me from changing the ?x86_64 to x86_64.

Example:
  * https://dev.haiku-os.org/ticket/16840 -- Crash on exit / deconstruct
  * https://dev.haiku-os.org/ticket/16847 -- Misrendering on non 32-bit colorspaces
  * BGLView is dark until window is moved / resized

**Comment by X547** [comment]
> BGLView is dark until window is moved / resized

Any bug reports/way to reproduce for this?

**Comment by waddlesplash** [comment]
I think it occurs in most applications is what @kallisti5 was saying; GLTeapot was affected.

**Comment by kallisti5** [comment]
err.. don't you need the patchset?  The build  fix missing from 21.0.0 is also missing in 21.0.1

**Comment by kenmays** [comment]
This one is included the the current 21.0.1 patchset.

File: sys-libs/mesa/mesa-21.0.1.recipe [hunk 1]
```diff
@@ -7,13 +7,12 @@ HOMEPAGE="https://www.mesa3d.org/"
 COPYRIGHT="1999-2021 Brian Paul"
 LICENSE="MIT"
 REVISION="1"
-#SOURCE_URI="https://mesa.freedesktop.org/archive/mesa-${portVersion}.tar.xz"
-SOURCE_URI="https://mesa.freedesktop.org/archive/mesa-21.0.0-rc5.tar.xz"
-SOURCE_DIR="mesa-21.0.0-rc5"
-CHECKSUM_SHA256="f57d23aa69d5ed7cfde250a7bf8f72285a34692f9e8d541532fa6970f941ce01"
+SOURCE_URI="https://mesa.freedesktop.org/archive/mesa-${portVersion}.tar.xz"
+SOURCE_DIR="mesa-${portVersion}"
+CHECKSUM_SHA256="379fc984459394f2ab2d84049efdc3a659869dc1328ce72ef0598506611712bb"
```
Comment (Begasus) [inline sys-libs/mesa/mesa-21.0.1.recipe:12]: Switch places:
```diff
-SOURCE_DIR="mesa-${portVersion}"
CHECKSUM_SHA256="379fc984459394f2ab2d84049efdc3a659869dc1328ce72ef0598506611712bb"
+SOURCE_DIR="mesa-${portVersion}"
```

Comment (kenmays) [reply sys-libs/mesa/mesa-21.0.1.recipe]: fixed

**Comment by kenmays** [comment]
Tested:
Apps:
[ ] GLInfo - see: https://dev.haiku-os.org/ticket/16840 -- Crash on exit / deconstruct
[ ] Haiku3D
Sxreensavers:
[ ] Flurry
[ ] GLife
[ ] Gravity

**Comment by kallisti5** [comment]
I submitted https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/10135 to Mesa to try and get the build fix into the 21.0 branch.

**Comment by kenmays** [comment]
Note: All of the recent Haiku Mesa patches were merged in the Mesa 21.0.3 release as of Apr 24, 2021.

**Pull request merged**